onnx/checker.cc

```diff
@@ -1036,7 +1036,7 @@ void check_model(const std::string& model_path, bool full_check, bool skip_opset
   check_model(model, ctx);
 
   if (full_check) {
-    ShapeInferenceOptions options{true, 1, false};
+    ShapeInferenceOptions options{true, FailAnyInferenceError, false};
```

Comment:
Would it be possible to scope this enum for clarity?

Reply:
that is correct, this PR is far from complete. There are many changes on API surface so we want to make it right before merge. I am more concerned on the Python side. Naming is one thing that we need to be precise. In terms of scope, it is already under ShapeInference module but adding errors.
It is less a concern on the c++ side however.

Reply:
Curious why it's less of a concern on the c++ side?

Reply:
I should have said it was a concern for both. It is just that python api is more visible.

Reply:
@justinchuby it is better to scope the enum. do you think InferenceErrorMode instead of error is better?

enum class InferenceErrorMode : uint32_t {
  IgnoreInferenceError, // Ignore any inference errors
  FailAnyInferenceError,  // Fail on any inference error
  FailShapeInferenceError,  // Fail on any shape inference error, like merging existing shape with inferred etc.
  FailTypeInferenceError  // Fail on any type inference error
};